back/engines/commercial/analytics/app/services/analytics/populate_dimensions_service.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Analytics
+  class PopulateDimensionsService
+    class << self
+      def run
+        populate_dates
+        populate_types
+        populate_locales
+        populate_referrer_types
+      end
+
+    private
+
+      def populate_dates
+        dates_count = Analytics::DimensionDate.count
+        from = Time.zone.today
+        to = Time.zone.today + 6.months
+
+        if dates_count == 0
+          first_idea = Idea.order(:created_at).limit(1).pluck(:created_at)
+          first_initiative = Initiative.order(:created_at).limit(1).pluck(:created_at)
+          first_post = [first_idea, first_initiative].min[0]
+          unless first_post.nil?
+            from = first_post.to_datetime
+          end
+        else
+          last_date = Analytics::DimensionDate.order(date: :desc).limit(1).pluck(:date)
+          from = last_date[0].to_datetime + 1.day
```

Comment:
```suggestion
          last_date = Analytics::DimensionDate.order(date: :desc).first.date
          from = last_date + 1.day
```

(Is the convertion to datetime needed?)